### ISSUE TITLE
Get account data from another app via intent

### ISSUE THREAD

**Issue by sw-dev1** [opacapp/opacclient#465]
In the manifest there is an intent filter 
```
<intent-filter>
                <action android:name="de.geeksfactory.opacclient.ACCOUNT"/>
                <category android:name="android.intent.category.DEFAULT"/>
  </intent-filter>

```
Is it intended (pun unintended....) to return the account data to another app via an startActivityForResult?

**Comment by johan12345** [comment]
Not exactly. It can be used to link to the account page in this app for a specific library (if you have an account set up for that library). There's a similar intent for search, too.

It was intended to be used by an app showing a map of all the libraries, which as far as I know is not released yet.

An Intent that returns the actual account data to another app would be an interesting idea, but one would need to be careful not to create a security issue by implementing that.

**Comment by raphaelm** [comment]
> An Intent that returns the actual account data to another app would be an interesting idea, but one would need to be careful not to create a security issue by implementing that.

First of all, wouldn't that rather be a ContentProvider than an Intent?

Second, I'm really against providing that (except maybe with a opt-in setting to enable it) as it is a huge privacy problem otherwise.

**Comment by sw-dev1** [comment]
@raphaelm Absolutely a ContenProvider would be ideal. Concerning privacy, the data remains within the device and the user of the device has control to which app he/she grants permission to this contentprovider (the other app would need to declare it in the manifest). It does not seem to be more of a privacy concern than e.g. the contacts or calendar provider. They provide read and even write access to the databases.

**Comment by sw-dev1** [comment]
Talking about the calendar, what do you think about a feature that would open the calendar from the notification about expiring items to allow the user to plan their next visit to the library?

**Comment by raphaelm** [comment]
About the privacy: No, the permission wouldn't be enough for me, looking at how coarse the user interface for permission granting is. I don't think that is good enough for the user to make an informed decision. If the data "remains on the device" is not something we control, the other apps can do with it what they want. The content provider would need to be disabled (as in: returning empty results) by default and only be activated by a setting.

**Comment by raphaelm** [comment]
Regarding the calendar: Sure, would be useful. However, our notification already includes two action buttons, I'm not sure if more are feasible.
